### PR TITLE
MESOS: Fix broken km and hypercube due to too early proxy initialization

### PR DESCRIPTION
**Pull request by sttts**
This was introduced in 1c25c2cd9954d2985d115e3dc876918222aff7db.

Fixes https://github.com/mesosphere/kubernetes-mesos/issues/514

### CODE REVIEW

File: contrib/mesos/cmd/km/kube-proxy.go [hunk 1]
```diff
@@ -40,13 +40,14 @@ func NewKubeProxy() *Server {
...
-	s, err := kubeproxy.NewProxyServerDefault(config)
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
-	}
 
 	hks.Run = func(_ *Server, args []string) error {
+		s, err := kubeproxy.NewProxyServerDefault(config)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%v\n", err)
```
Comment (jdef) [inline contrib/mesos/cmd/km/kube-proxy.go]: just return `err` here instead of printing/exiting

**Comment by jdef** [comment]
It may be worth submitting an additional PR for fixing the upstream hyperkube too

**Comment by jdef** [comment]
minor comment, otherwise lgtm

File: contrib/mesos/cmd/km/kube-proxy.go (hunk 1 above)
Comment (sttts) [inline contrib/mesos/cmd/km/kube-proxy.go]: right

**Comment by sttts** [comment]
@gmarek please take a look. We only tested this for the Mesos km. The second commit does the same for the hypercube.

Do we miss anything essential by not calling `NewProxyServerDefault` early?

**Comment by gmarek** [comment]
I believe someone broke the head again. Let me check.

**Comment by sttts** [comment]
@k8s-bot test this please

**Comment by brosner** [comment]
I can confirm the change to hyperkube is the correct fix. It has been working for me in my development cluster.

**Comment by gmarek** [comment]
@k8s-bot test this

**Pull request merged**